packages/eip-1193-provider/src/utils.ts

```
@@ -123,7 +122,8 @@ export const validateChain = (
   const { rpcUrls, blockExplorerUrls, chainId, nativeCurrency } = chain;
 
   if (addedChains.some((c) => c.chainId === chainId)) {
```

can't we just remove this?

this is opinionated, but IMO if someone passes a `chainId` that already exists, but with an invalid RPC, we should throw an error about the RPC being invalid

right now, it'll silently succeeds without any action